src/nvim/undo.c

```diff
@@ -3146,11 +3176,13 @@ u_header_T *u_force_get_undo_header(buf_T *buf)
     buf_T *save_curbuf = curbuf;
     curbuf = buf;
     // Args are tricky: this means replace empty range by empty range..
-    u_savecommon(0, 1, 1, true);
+    u_savecommon(curbuf, 0, 1, 1, true);
+    curbuf = save_curbuf;
```

Comment:
this doesn't look right. perhaps this `save_curbuf` declaration can be eliminated now?